pydantic_ai_slim/pydantic_ai/mcp.py

```diff
@@ -303,6 +450,87 @@ def tool_for_tool_def(self, tool_def: ToolDefinition) -> ToolsetTool[Any]:
             args_validator=TOOL_SCHEMA_VALIDATOR,
         )
 
+    async def list_resources(self) -> list[Resource]:
+        """Retrieve resources that are currently present on the server.
+
+        Note:
+        - We don't cache resources as they might change.
+        - We also don't subscribe to resource changes to avoid complexity.
+
+        Raises:
+            MCPError: If the server returns an error.
+        """
+        async with self:  # Ensure server is running
+            if not self.capabilities.resources:
+                return []
+            try:
+                result = await self._client.list_resources()
+            except mcp_exceptions.McpError as e:
+                raise MCPError.from_mcp_sdk_error(e) from e
```

Comment:
Can we have a test on the error output?

Also, on our side, I think we should be more specific on the errors, `MCPError` should be the parent of the other errors, e.g. in this case `ResourceNotFound(MCPError)` or `ResourceNotSupported(MCPError)` (?).